typing_extensions/README.rst

```diff
@@ -9,15 +9,19 @@ Typing Extensions
 Overview
 ========
 
-The ``typing_extensions`` module contains both backports of ``typing`` features
-as well as experimental types that will eventually be added to the ``typing``
-module, such as ``Protocol`` (see PEP 544 for details about protocols and
-static duck typing) or ``TypedDict`` (see PEP 589).
-
-Users of other Python versions should continue to install and use
-the ``typing`` module from PyPi instead of using this one unless
-specifically writing code that must be compatible with multiple Python
-versions or requires experimental types.
+The ``typing_extensions`` module serves two related purposes:
+
+- Enable use of new type system features on older Python versions. For example,
+  ``typing.TypeGuard`` is new in Python 3.10, but ``typing_extensions`` allows
+  users on Python 3.6 through 3.9 to use it too.
+- Enable experimentation with new type system PEPs before they are accepted and
+  added to the ``typing`` module.
+  
+New features may be added to ``typing_extensions`` as soon as they are specified
+in a PEP that has been added to the `python/peps <https://github.com/python/peps>`_
+repository. If the PEP is accepted, the feature will then be added to ``typing``
+for the next CPython release. No typing PEP has been rejected so far, so we
+haven't yet figured out how to deal with that possibility.
```

Comment:
lol

Reply:
Maybe we'll remove it in the next major version. But I have no strong feelings regarding this.

Reply:
That makes sense to me too, but we can figure that out when we get to it.